snapshot.go

```diff
@@ -77,3 +78,113 @@ func (c *Client) UpdateSnapshot(snapshotID string, request SnapshotProperties) (
 	err := c.Patch(url, request, ret, http.StatusAccepted)
 	return ret, err
 }
+
+// DeleteSnapshotAndWait deletes a specified snapshot and waits for the request
+// to complete.
+func (c *Client) DeleteSnapshotAndWait(snapshotID string) error {
+	ret, err := c.DeleteSnapshot(snapshotID)
+	if err != nil {
+		return err
+	}
+	return c.WaitTillProvisioned(ret.Get("location"))
+}
+
+// ListSnapshotsWithSelector retrieves all snapshots and performs client-side
+// filtering according to the list of selectors. Each selector is concatenated
+// with logical AND.
+func (c *Client) ListSnapshotsWithSelector(selectors ...SnapshotSelector) ([]Snapshot, error) {
+	url := snapshotsPath()
+	ret := &Snapshots{}
+	err := c.Get(url, ret, http.StatusOK)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []Snapshot
+outerLoop:
+	for _, snapshot := range ret.Items {
+		for _, selector := range selectors {
```

Comment:
you could probably use SelectExactSnapshot here?

Reply:
Actually now that I think about it, I don't know if SelectExactSnapshot should even exist if it is the default behaviour. Wdyt?

Reply:
well, I like the idea to build complex filters with this. `SelectAnySnapshot(SelectExactSnapshot(ByName(),ByDescription()),ByState()))`